conda-store-server/conda_store_server/orm.py

```diff
@@ -190,15 +194,20 @@ def build_path(self, conda_store):
         """
         store_directory = os.path.abspath(conda_store.store_directory)
         namespace = self.environment.namespace.name
-        name = self.specification.name
-        return (
+        res = (
             pathlib.Path(
                 conda_store.build_directory.format(
-                    store_directory=store_directory, namespace=namespace, name=name
+                    store_directory=store_directory,
+                    namespace=namespace,
                 )
             )
             / self.build_key
         )
+        # conda prefix must be less or equal to 255 chars
+        # https://github.com/conda-incubator/conda-store/issues/649
+        if len(str(res)) > 255:
+            raise BuildPathError("build_path too long: must be <= 255 characters")
```

Comment:
Maybe we can afford to be more informative here and report which components are responsible for `build_path`. What about:

```python
f"'build_path' is too long ({len(str(build_path))}). Maximum allowed length is 255. See docs at xxx for more information.
```

This is also an opportunity to start codifying error numbers and have a reference online so folks can troubleshoot, à la https://flake8.pycqa.org/en/latest/user/error-codes.html. (worth its own issue).

Reply:
That what I had originally, but I removed it. See here https://github.com/conda-incubator/conda-store/pull/653#issuecomment-1807184045. As for codifying the errors: not adding that here either. There's no policy/format on this right now. Similarly, saying "see the docs" is not helpful. This also needs to be linked and done in a systematic way, otherwise links will point to nowhere over time. Not adding this also because there's no infra for this right now. Let's keep the PR self-contained, we can always change this later.

Reply:
> Let's keep the PR self-contained, we can always change this later.

Works for me, but let's make sure there's an issue we can iterate on.